### PR TITLE
[ANCHOR-828] Improve PropertyQueueConfig validations

### PR DESCRIPTION
**Pull request by lijamie98**
### Description

- Merge `PropertyEventConfig` and `PropertyQueueConfig` validations
- Bug: `PropertyQueueConfig` validation was not invoked.
- Improved the error message.
- Improved the `ssl_keystore_location` and `ssl_truststore_location` checks

### Context

- Validation bug fixes
- Validation error message improvements.

### Testing

- `./gradlew test`

### Documentation

N/A

- Attach stellar-docs pull request, documenting new feature
- If it's an urgent feature request, please create a ticket and attach ticket number to this PR

### Known limitations

 N/A

### CODE REVIEW

**Review by JakeUrban — APPROVED** [review]

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertyQueueConfig.java [hunk 1]
```diff
@@ -44,109 +45,116 @@ public void validate(@NotNull Object target, @NotNull Errors errors) {
...
-              "queue.kafka.sasl_mechanism must be defined if securityProtocol is not PLAINTEXT.");
+              "events.queue.kafka.sasl_mechanism must be defined when securityProtocol is SASL_PLAINTEXT.");
         }
       }
 
       if (kafkaConfig.getSecurityProtocol() == KafkaConfig.SecurityProtocol.SASL_SSL) {
+        if (kafkaConfig.getSaslMechanism() == null) {
+          errors.reject(
+              "kafka-sasl-mechanism-empty",
+              "events.queue.kafka.sasl_mechanism must be defined when securityProtocol is SASL_SSL.");
```
Comment (philipliu) [inline platform/src/main/java/org/stellar/anchor/platform/config/PropertyQueueConfig.java:101]:
```suggestion
              "events.queue.kafka.sasl_mechanism must be defined when queue.kafka.security_protocol is SASL_SSL.");
```

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertyQueueConfig.java [hunk 2]
```diff
@@ -44,109 +45,116 @@ public void validate(@NotNull Object target, @NotNull Errors errors) {
...
+      }
+
+      if (kafkaConfig.getSecurityProtocol() == KafkaConfig.SecurityProtocol.SASL_PLAINTEXT) {
         if (kafkaConfig.getSaslMechanism() == null) {
-          errors.rejectValue(
-              "kafka.saslMechanism",
+          errors.reject(
               "kafka-sasl-mechanism-empty",
-              "queue.kafka.sasl_mechanism must be defined if securityProtocol is not PLAINTEXT.");
+              "events.queue.kafka.sasl_mechanism must be defined when securityProtocol is SASL_PLAINTEXT.");
```
Comment (philipliu) [inline platform/src/main/java/org/stellar/anchor/platform/config/PropertyQueueConfig.java:93]:
```suggestion
              "events.queue.kafka.sasl_mechanism must be defined when queue.kafka.security_protocol is SASL_PLAINTEXT.");
```

File: platform/src/test/kotlin/org/stellar/anchor/platform/config/EventConfigTest.kt [hunk 3]
```diff
@@ -8,23 +8,23 @@ import org.junit.jupiter.params.ParameterizedTest
...
 import org.junit.jupiter.params.provider.MethodSource
 import org.springframework.validation.BindException
-import org.springframework.validation.Errors
 import org.springframework.validation.ValidationUtils
 import org.stellar.anchor.config.event.QueueConfig.QueueType.*
 
 class EventConfigTest {
   lateinit var config: PropertyEventConfig
-  lateinit var errors: Errors
+  //  lateinit var errors: Errors
```
Comment (philipliu) [inline platform/src/test/kotlin/org/stellar/anchor/platform/config/EventConfigTest.kt:16]: Meant to uncomment?

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertyQueueConfig.java [hunk 4]
```diff
@@ -44,109 +45,116 @@ public void validate(@NotNull Object target, @NotNull Errors errors) {
         validateMsk(config, errors);
         break;
       default:
-        errors.rejectValue(
-            "type", "invalidType-queue-type", "queue.type must be one of 'KAFKA', 'SQS', or MSK");
+        errors.reject("invalidType-queue-type", "queue.type must be one of 'KAFKA', 'SQS', or MSK");
     }
   }
 
-  private void validateMsk(PropertyQueueConfig ignoredConfig, Errors errors) {
```
Comment (philipliu) [inline platform/src/main/java/org/stellar/anchor/platform/config/PropertyQueueConfig.java:52]: Why are we removing these validations? Doesn't this imply we support them?

**Review by philipliu — COMMENTED** [review]
Does this fix this exception when keystore and truststore are missing?

```
java.io.FileNotFoundException: class path resource [] cannot be resolved to absolute file path because it does not reside in the file system: jar:file:/app/anchor-platform-runner.jar!/BOOT-INF/classes!/
```

**Pull request merged**